apps/voting-disputable/test/contracts/voting_execution.js

```diff
@@ -22,98 +22,168 @@ contract('Voting', ([_, owner, holder20, holder29, holder51]) => {
+          it('cannot be executed immediately executed', async () => {
+            await assertRevert(voting.executeVote(voteId), VOTING_ERRORS.VOTING_CANNOT_EXECUTE)
+          })
+
+          it('cannot execute vote if not enough quorum met', async () => {
+            await voting.vote(voteId, true, { from: holder20 })
+            await voting.mockIncreaseTime(VOTE_DURATION)
+
+            await assertRevert(voting.executeVote(voteId), VOTING_ERRORS.VOTING_CANNOT_EXECUTE)
+          })
+
+          it('cannot execute vote if not support met', async () => {
+            await voting.vote(voteId, false, { from: holder29 })
+            await voting.vote(voteId, false, { from: holder20 })
+            await voting.mockIncreaseTime(VOTE_DURATION)
+
+            await assertRevert(voting.executeVote(voteId), VOTING_ERRORS.VOTING_CANNOT_EXECUTE)
+          })
+
+          it('can be executed if quorum and support are met', async () => {
+            await voting.vote(voteId, true, { from: holder51 })
+            await voting.mockIncreaseTime(VOTE_DURATION)
+
+            await voting.executeVote(voteId)
+
+            assertBn(await executionTarget.counter(), expectedExecutions, 'execution times do not match')
+          })
+
+          it('cannot be executed twice', async () => {
+            await voting.vote(voteId, true, { from: holder51 })
+            await voting.mockIncreaseTime(VOTE_DURATION)
+
+            await voting.executeVote(voteId)
+
+            await assertRevert(voting.executeVote(voteId), VOTING_ERRORS.VOTING_CANNOT_EXECUTE)
+          })
+        }
+
+        context('with a single action script', () => {
+          itCanBeExecuted(1)
         })
 
-        it('can be executed if quorum and support are met', async () => {
-          await voting.vote(voteId, true, { from: holder51 })
-          await voting.mockIncreaseTime(VOTE_DURATION)
-          await voting.executeVote(voteId)
+        context('with a multiple action script', () => {
+          itCanBeExecuted(3)
+        })
+      })
 
-          assertBn(await executionTarget.counter(), expectedExecutions, 'execution times do not match')
+      context('with a failing script', async () => {
+        beforeEach('create script', async () => {
+          ({ script } = await voteScript())
+          script = script.slice(0, -2) // remove one byte from calldata for it to fail
         })
 
-        it('cannot be executed twice', async () => {
+        it('reverts', async () => {
+          ({ voteId } = await createVote({ voting, script, from }))
+
           await voting.vote(voteId, true, { from: holder51 })
           await voting.mockIncreaseTime(VOTE_DURATION)
-          await voting.executeVote(voteId)
 
-          await assertRevert(voting.executeVote(voteId), VOTING_ERRORS.VOTING_CANNOT_EXECUTE)
+          await assertRevert(voting.executeVote(voteId), 'EVMCALLS_INVALID_LENGTH')
```

Comment:
Can we add this one to the error constants file?

Reply:
Added in https://github.com/aragon/aragon-apps/pull/1195/commits/3b0d3a0d4cb74db5d6af759e10237cffa3d4d3d2